CMakePresets.json

```diff
@@ -0,0 +1,216 @@
+        "VAST_ENABLE_BUNDLED_CAF": "ON",
+        "VAST_PLUGINS": "${sourceDir}/plugins/pcap;${sourceDir}/plugins/sigma;${sourceDir}/plugins/summarize"
+      }
+    },
+    {
+      "name": "nix-clang-redeb",
+      "inherits": ["nix-base", "clang-base"],
+      "displayName": "Clang Redeb Config",
+      "binaryDir": "${sourceDir}/build/clang/redeb",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/clang/redeb",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
+      }
+    },
+    {
+      "name": "nix-clang-release",
+      "inherits": ["nix-base", "clang-base"],
+      "displayName": "Clang Release Config",
+      "binaryDir": "${sourceDir}/build/clang/release",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/clang/release",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
+      }
+    },
+    {
+      "name": "gcc-base",
+      "hidden": true,
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "gcc",
+        "CMAKE_CXX_COMPILER": "g++"
+      }
+    },
+    {
+      "name": "nix-gcc-debug",
+      "inherits": ["nix-base", "gcc-base"],
+      "displayName": "GCC Debug Config",
+      "binaryDir": "${sourceDir}/build/gcc/debug",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/gcc/debug",
+        "VAST_ENABLE_ASAN": "ON",
+        "VAST_ENABLE_BUNDLED_CAF": "ON",
+        "VAST_PLUGINS": "${sourceDir}/plugins/pcap;${sourceDir}/plugins/sigma;${sourceDir}/plugins/summarize"
+      },
+      "environment": {
+        "LDFLAGS": "-fuse-ld=lld"
+      }
+    },
+    {
+      "name": "nix-gcc-redeb",
+      "inherits": ["nix-base", "gcc-base"],
+      "displayName": "GCC Redeb Config",
+      "binaryDir": "${sourceDir}/build/gcc/redeb",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/gcc/redeb",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
+      },
+      "environment": {
+        "LDFLAGS": "-fuse-ld=lld"
+      }
+    },
+    {
+      "name": "nix-gcc-release",
+      "inherits": ["nix-base", "gcc-base"],
+      "displayName": "GCC Release Config",
+      "binaryDir": "${sourceDir}/build/gcc/release",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/gcc/release",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
+      },
+      "environment": {
+        "LDFLAGS": "-fuse-ld=gold"
+      }
+    },
+    {
+      "name": "nix-static",
+      "inherits": ["nix-base"],
+      "hidden": true,
+      "cacheVariables": {
+        "BUILD_SHARED_LIBS": "OFF",
+        "BUILD_TESTING": "OFF",
+        "CMAKE_C_COMPILER": "x86_64-unknown-linux-musl-gcc",
+        "CMAKE_CXX_COMPILER": "x86_64-unknown-linux-musl-g++",
+        "CMAKE_SYSTEM_NAME": "Linux",
+        "CMAKE_SYSTEM_PROCESSOR": "x86_64",
```

Comment:
Why do we need to set these two manually?

Reply:
Static binaries only make sense for Linux, but you're right we probably shouldn't hard-code `x86_64`.

Reply:
We should probably add something like this to our CMake then.

```cmake
if (VAST_ENABLE_STATIC_BINARY AND CMAKE_SYSTEM_NAME NOT EQUAL "Linux")
  message(FATAL_ERROR ...)
endif ()
```